src/Integration/LocalServices/TestProjectIndicators/ServiceGuidTestProjectIndicator.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using System.Xml.Linq;
+using System.Xml.XPath;
+using EnvDTE;
+
+namespace SonarLint.VisualStudio.Integration.LocalServices.TestProjectIndicators
+{
+    public class ServiceGuidTestProjectIndicator : ITestProjectIndicator
+    {
+        private readonly IFileSystem fileSystem;
+        private const string TestServiceGuid = "{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}";
+
+        public ServiceGuidTestProjectIndicator()
+            : this(new FileSystem())
+        {
+        }
+
+        internal ServiceGuidTestProjectIndicator(IFileSystem fileSystem)
+        {
+            this.fileSystem = fileSystem ?? throw new ArgumentNullException(nameof(fileSystem));
+        }
+
+        public bool? IsTestProject(Project project)
+        {
+            var projectPath = project.FullName;
```

Comment:
It's probably worth guarding against file IO exceptions, especially if this code is ever called on the main UI thread. It might be simplest to catch and suppress all non-critical exceptions.

Reply:
FYI there a method `ErrorHandler.IsCriticalException` that's used to decide whether an exception should be suppressed or not. The usual pattern is like this:
```
catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
```

I'll push a commit with this fix and then I'll merge the PR.